docs/input_output.md

```diff
@@ -2,85 +2,279 @@
+          "provider": "<string>"    // CMR provide ID
+        },
+        "provider": {            // Provider configuration information taken from the 'providers' of 
+                                 // collection configuration.  Any fields are allowed.
+          "id": "<string>",      // An id used to identify this provider
+          "anykey": "anyvalue"   // additional configuration items
+        },
+        "collection": {          // Metadata taken from the 'meta' attribute of the collection. These can 
+                                 // contain any data that is specific to the collection.
+          "anykey": "anyvalue"
+        },
+        "queues": {              // List of SQS queues that are used by the cumulus-api
+          "anyqueue": "queue endpoint"
+        }
+      },
+      "workflow_config": {       // Defines configuration for tasks that are part of a workflow as a map 
+                                 // of task name to a JSON object containing configuration settings
+        "any task name": "any ask configuration" // task configuration for each task in the workflow
+      },
+      "exception": "<object>",   // An optional field that a task can return with any valid as a signal 
+                                 // that the task aborted for a reason. Information in this can be used by 
+                                 // the workflow to determine next steps
+      "payload": "<object>"      // A Task-specific payload. This can be any data type required by the Task.
+                                 // It can be considered the input and output of the Task, whereas the other 
+                                 // fields are execution context. Tasks should document their expected payload 
+                                 // input and output formats. Generally a Task will return an object which is 
+                                 // nearly identical to its input in all fields but 'payload', and 'payload' 
+                                 // will be completely different"
+    }
+
+## Cumulus Message Adapter
 
-The Scheduler service creates the initial event by combining the collection configuration (See [collections.json](../config/collections.json)) with the AWS execution context provided by its [CloudFormation template](../config/cloudformation.yml.erb).  The collection configuration supports inheritance. If the `"parent"` attribute of a collection references the `"id"` of another collection, the other collection's fields will be recursively copied into the collection.
+The Scheduler service creates the initial event by combining the collection configuration, external resource configuration, workflow configuration, and deployment environment settings.  The subsequent workflow messages between tasks must conform to the message schema. By using Cumulus Message Adapter, individule task Lambda function can focus on the input and output specific to the task, and does not worry about the non-task related message fields.
```

Comment:
individual task Lambda functions only receive the input and output specifically configured for the task, and not non-task-related message fields.